sound/soc/sof/ipc4-topology.c

```diff
@@ -679,11 +693,24 @@ static int sof_ipc4_widget_setup_comp_pga(struct snd_sof_widget *swidget)
 
 	return 0;
 err:
+	sof_ipc4_free_audio_fmt(&gain->available_fmt);
 	kfree(gain);
 	swidget->private = NULL;
```

Comment:
optimization: you could replace those three lines by a call to optimization: you could replace those three lines by a call to sof_ipc4_widget_free_comp_pga()